rpc-client-api/src/client_error.rs

```diff
@@ -0,0 +1,199 @@
+pub use reqwest;
+use {
+    crate::{rpc_request, rpc_response},
+    solana_faucet::faucet::FaucetError,
+    solana_sdk::{
+        signature::SignerError, transaction::TransactionError, transport::TransportError,
+    },
+    std::io,
+    thiserror::Error,
+}; // export `reqwest` for clients
+
+#[derive(Error, Debug)]
+pub enum ClientErrorKind {
+    #[error(transparent)]
+    Io(#[from] io::Error),
+    #[error(transparent)]
+    Reqwest(#[from] reqwest::Error),
+    #[error(transparent)]
+    RpcError(#[from] rpc_request::RpcError),
+    #[error(transparent)]
+    SerdeJson(#[from] serde_json::error::Error),
+    #[error(transparent)]
+    SigningError(#[from] SignerError),
+    #[error(transparent)]
+    TransactionError(#[from] TransactionError),
+    #[error(transparent)]
+    FaucetError(#[from] FaucetError),
```

Comment:
TransactionError and FaucetError, both?

Reply:
Yes, FaucetError can definitely be removed, since airdrop requests all go through RPC now instead of communicating with the faucet directly.
I'll have to see about TransactionError.

Reply:
We definitely use the TransactionError variant pretty heavily in rpc_client `send_and_confirm...` methods, bubbled up from signature statuses themselves.
So I'm leaving untouched for now, and we can consider it as part of a ClientErrorKind-rework follow-up